[SecurityBundle] Remove autoconfiguration for `ProcessorInterface`

derrabus opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 6.1
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | N/A
| License       | MIT
| Doc PR        | N/A

This PR removes autoconfiguration logic for a `Symfony\Bridge\Monolog\Processor\ProcessorInterface` that does not exist. As far as I can tell, this has always been unused code, so we might as well remove it.

derrabus committed:
Remove autoconfiguration for ProcessorInterface

nicolas-grekas commented:
I guess that's a wrong namespace, maybe related to history/renames?
See https://github.com/Seldaek/monolog/blob/main/src/Monolog/Processor/ProcessorInterface.php
Git blame will point at me for the removed lines, and I'm sure I added them for a reason (can't recall right now.)

derrabus commented:
> I guess that's a wrong namespace, maybe related to history/renames?

The pass was added in #33663 and those lines haven't changed since then. I've browsed the source tree that very commit but the bridge did not have that interface at that time either.

I dug deeper and apparently, the interface did exist for a brief moment: Introduced in #27801, reverted in #28845.

I assume that the lines are an artifact of an earlier iteration of that PR. Since they effectively never did anything, the fix for me is to remove them.

nicolas-grekas commented:
The purpose of those lines was to wire processors automatically with the untracked token storage, so that they could log things found in the session without triggering `Cache-Control: private` on the response.

We should fix this behavior of it's broken.

derrabus commented:
Should we build that logic in MonologBundle instead?

nicolas-grekas commented:
I was having a look, monolog bundle already registers the tag for autoconfiguration:
https://github.com/symfony/monolog-bundle/blob/fde12fc628162787a4e53877abadc30047fd868b/DependencyInjection/MonologExtension.php#L131-L134

What this does is the binding. And we could argue that the binding is for SecurityBundle.

To me, we should fix the namespace in the lowest affected branch.
